### PR TITLE
add service name to nextjs and remix

### PR DESCRIPTION
**Pull request by et**
## Summary

Allows service name to be set for nextjs and remix backends. Note: the frontend could already set this since `<HighlightInit />` uses `H.init` under the hood.

## How did you test this change?

Using the nextjs e2e app (`yarn run e2e:nextjs`), confirmed NextJS errors have a service name attached:

Screenshot 2023-08-31 at 2 33 34 PM
Screenshot 2023-08-31 at 2 34 17 PM
Screenshot 2023-08-31 at 2 34 20 PM

Using the remix e2e app (`yarn run e2e:remix`), confirmed Remix errors have a service name attached:
Screenshot 2023-08-31 at 2 35 30 PM
Screenshot 2023-08-31 at 2 38 27 PM

## Are there any deployment considerations?

Bumped versions for the remix and nextjs sdks

### CODE REVIEW

File: sdk/highlight-node/src/client.ts [hunk 1]
```diff
@@ -58,6 +58,7 @@ export class Highlight {
 
 		this.otel = new NodeSDK({
 			autoDetectResources: true,
+			resourceDetectors: [processDetectorSync],
```
Comment (et) [inline sdk/highlight-node/src/client.ts:61]: Should fix node processes for #6465. Will still need to validate other SDKs

File: sdk/highlight-node/src/client.ts [hunk 2]
```diff
@@ -184,7 +185,6 @@ export class Highlight {
 		if (metadata != undefined) {
 			span.setAttributes(metadata)
 		}
-		span.setAttributes({ ['highlight.project_id']: this._projectID })
```
Comment (et) [inline sdk/highlight-node/src/client.ts:187]: This is unnecessary to set at the span level. It's set at the resource level (see the constructor).

File: sdk/highlight-next/CHANGELOG.md [hunk 3]
```diff
@@ -50,3 +50,15 @@
 
 - Removing exports for `@highlight-run/next` and `@highlight-run/next/highlight-init`. Import from `@highlight-run/next/server` and `@highlight-run/next/client` instead.
 - Adding `excludedHostnames?: string[]` to `HighlightInit` props. Pass in a list of full or partial hostnames to prevent tracking: `excludedHostnames={['localhost', 'staging']}`.
+
+### 4.1.0
+
+### Minor changes
+
+- Added support for setting metadata on `consumeError`
```
Comment (et) [inline sdk/highlight-next/CHANGELOG.md:58]: Missing changelog entry for #6338

File: highlight.io/highlight.config.ts [hunk 4]
```diff
@@ -3,4 +3,5 @@ import { Highlight } from '@highlight-run/next/server'
 export const withHighlight = Highlight({
 	projectID: '4d7k1xeo',
 	debug: false,
+	serviceName: 'highlight.io',
```
Comment (et) [inline highlight.io/highlight.config.ts:6]: This won't have any affect until #6485 is fixed.

File: e2e/remix/package.json [hunk 5]
```diff
@@ -4,7 +4,7 @@
 	"sideEffects": false,
 	"scripts": {
 		"build": "remix build",
-		"dev": "remix dev",
+		"dev": "PORT=4352 remix dev",
```
Comment (et) [inline e2e/remix/package.json:7]: Remix runs on port 3000 by default which conflicts with our frontend app.

File: sdk/highlight-node/src/client.ts (hunk 1 above)
Comment (et) [reply sdk/highlight-node/src/client.ts:61]: Here are the created services with the process data populated correctly.

Screenshot 2023-08-31 at 2 40 13 PM


Frontend parts aren't created yet per #6486

File: e2e/remix/app/entry.server.tsx [hunk 6]
```diff
@@ -17,8 +17,14 @@ import { RemixServer } from '@remix-run/react'
...
 import isbot from 'isbot'
 import { renderToPipeableStream } from 'react-dom/server'
+import type { NodeOptions } from '@highlight-run/node'
 
-const nodeOptions = { projectID: CONSTANTS.HIGHLIGHT_PROJECT_ID }
+const nodeOptions: NodeOptions = {
+	projectID: CONSTANTS.HIGHLIGHT_PROJECT_ID,
+	otlpEndpoint: 'http://localhost:4318',
+	serviceName: 'my-remix-backend',
+	serviceVersion: '1.0.0',
```
Comment (et) [inline e2e/remix/app/entry.server.tsx:26]: cc @deltaepsilon - I updated the remix e2e app to point to a local environment

**Review by SpennyNDaJets — APPROVED** [review]
LGTM

**Pull request merged**